larpandoracontent/LArTrackShowerId/TrackShowerIdFeatureTool.cc

```diff
@@ -38,7 +42,26 @@ void ShowerFitFeatureTool::Run(SupportVectorMachine::DoubleVector &featureVector
     if (PandoraContentApi::GetSettings(*pAlgorithm)->ShouldDisplayAlgorithmInfo())
         std::cout << "----> Running Algorithm Tool: " << this->GetInstanceName() << ", " << this->GetType() << std::endl;
 
-    featureVector.push_back(CutClusterCharacterisationAlgorithm::GetShowerFitWidth(pAlgorithm, pCluster, m_slidingShowerFitWindow));
+    if (m_ratioVariables)
+    {
+        float ratio(-1.f);
+        try
+        {
+            const TwoDSlidingFitResult slidingFitResultLarge(pCluster, m_slidingLinearFitWindow, LArGeometryHelper::GetWireZPitch(this->GetPandora()));
+            float straightLineLength = (slidingFitResultLarge.GetGlobalMaxLayerPosition() - slidingFitResultLarge.GetGlobalMinLayerPosition()).GetMagnitude();
+            if (straightLineLength > std::numeric_limits<double>::epsilon())
+                ratio = (CutClusterCharacterisationAlgorithm::GetShowerFitWidth(pAlgorithm, pCluster, m_slidingShowerFitWindow))/straightLineLength;
+        }
+        catch (const StatusCodeException &)
+        {
+            ratio = -1.f;
+        }
+        featureVector.push_back(ratio);
+    }
+    else
+    {
+        featureVector.push_back(CutClusterCharacterisationAlgorithm::GetShowerFitWidth(pAlgorithm, pCluster, m_slidingShowerFitWindow));
```

Comment:
Some shared aspects here between the with/without ratioVariables code. There is good exception handling for the ratio variable case, but nothing for the other case. Would suggest calculate shower fit width in same way (with exception handling) regardless of ratioVariables option. Then, for ratio option only, obtain straight line length (with exception handling) and divide.

Reply:
ok, I'll have a look